R/travis-auth.R

```diff
@@ -1,75 +1,69 @@
-auth_travis_ <- function(gh_token = NULL) {
-  cli::cat_bullet(
-    bullet = "pointer", bullet_col = "yellow",
-    " Authenticating to GitHub."
-  )
-  if (is.null(gh_token)) {
-    # Do not allow caching this token, it needs to be fresh
-    gh_token <- auth_github_(
-      "read:org", "user:email", "repo_deployment", "repo:status",
-      "read:repo_hook", "write:repo_hook"
+auth_travis <- function(endpoint = ".org") {
+  yml <- tryCatch({
+    readLines("~/.travis/config.yml")
+  },
+  warning = function(cond) {
+    cli::cat_bullet(
+      bullet = "pointer", bullet_col = "yellow",
+      c(
+        "To interact with the Travis CI API, an API token is required.",
+        "This is a one-time procedure. The token will be stored in your home directory in the '.travis' directory."
+      )
     )
+    message("Querying API token...")
+    utils::browseURL(sprintf("https://travis-ci%s/account/preferences", endpoint))
+    wait_for_clipboard_token(endpoint = endpoint)
+    return(readLines("~/.travis/config.yml"))
   }
-  auth_travis_data <- list(
-    "github_token" = gh_token$credentials$access_token
   )
-  auth_travis <- TRAVIS_POST(
-    url = "/auth/github",
-    body = auth_travis_data,
-    httr::user_agent("Travis/1.0"),
-    token = NULL
-  )
-  httr::stop_for_status(auth_travis, "authenticate with travis")
-  httr::content(auth_travis)$access_token
+
+  # create api token if none is found but config file exists
+  if (!any(grepl("token", yml))) {
+    requireNamespace("utils", quietly = TRUE)
+    utils::browseURL(sprintf("https://travis-ci%s/account/preferences", endpoint))
+    wait_for_clipboard_token(endpoint = endpoint)
+  }
 }
 
-#' Authenticate with Travis CI
-#'
-#' @description
-#' Authenticates with Travis using your Github account. Returns an access token.
-#' The token will be obtained only once in each
-#' R session, but it will never be cached on the file system.
-#' In most scenarios, these functions will be called implicitly by other functions.
-#'
-#' `auth_travis()` only performs the authentication with Travis CI.
-#'
-#' @param gh_token `[Token2.0]`\cr
-#'   A GitHub token, by default obtained from [auth_github()] using the
-#'   "read:org", "user:email", "repo_deployment", "repo:status",
-#'   "read:repo_hook", and "write:repo_hook" scopes.
-#'
-#' @export
-auth_travis <- memoise::memoise(auth_travis_)
+wait_for_clipboard_token <- function(endpoint) {
+  cli::cat_bullet(
+    bullet = "info", bullet_col = "yellow",
+    " Waiting for API token to appear on the clipboard."
```

Comment:
The user needs to paste now?

Reply:
Yes. The previous approach did not work reliably on my Linux machine and annoyed me quite a bit at some point.
This is based on my tiling-window manager behavior most likely and might be niche.
I do not have a strong opinion on that, we can revert it.